resources/helm/overlays/istio-telemetry/grafana/templates/ingress.yaml

```diff
@@ -1,21 +1,29 @@
 {{- if .Values.grafana.ingress.enabled -}}
+{{- range $index, $host := .Values.grafana.ingress.hosts }}
 apiVersion: route.openshift.io/v1
 kind: Route
 metadata:
-  name: grafana
-  namespace: {{ .Release.Namespace }}
+  name: grafana{{ if gt $index 0 }}-{{ $index }}{{ end }}
+  namespace: {{ $.Release.Namespace }}
   labels:
     app: grafana
-    release: {{ .Release.Name }}
+    release: {{ $.Release.Name }}
+{{- if gt (len $.Values.grafana.ingress.annotations) 0 }}
   annotations:
-    {{- range $key, $value := .Values.grafana.ingress.annotations }}
-      {{ $key }}: {{ $value | quote }}
-    {{- end }}
+{{- range $key, $value := $.Values.grafana.ingress.annotations }}
+    {{ $key }}: {{ $value | quote }}
+{{- end }}
+{{- end }}
 spec:
+{{- if ne $host "grafana.local" }}
+  host: {{ $host }}
+{{- end }}
```

Comment:
Found the following bug when I manually tested this. If I create an SMCP with:
```
spec:
  addons:
    grafana:
      enabled: true
      install:
        service:
          ingress:
            enabled: true
            hosts:
            - grafana-host.example.com
```
a route with the host `grafana-host.example.com` is created as expected. However, if I then edit the SMCP and remove the `hosts` field, the host in the route is still `grafana-host.example.com`. This Route doesn't match the one that's created if I initially create the SMCP without the `hosts` field:
```
spec:
  addons:
    grafana:
      enabled: true
      install:
        service:
          ingress:
            enabled: true
            hosts:
            - grafana-host.example.com
```
In this case, the host in the Route is `grafana-istio-system.apps.mluksa-mesh1.servicemesh....com`. 

Since the input state (SMCP spec) during the reconciliation of the SMCP is the same in both cases, the output (the Route) should also be the same, regardless of how we got to the input state.

The reason this happens is because you don't set the `host` field here, which means that the PATCH operation doesn't update it.

Reply:
I've been trying to find a solution for this, but it looks like there's no way to remove the host. I tried setting it to `nil`, `''`, and removing it with JSONPatch (`op:remove`). Nothing worked.

So, it looks like the only way to fix this would be to delete and recreate the Route.

Reply:
Hmm. That's a problem. We also can't default to a `recreate` strategy for Route objects in general as that would change the generated host name

Reply:
Okay, I added some code to the Patch Factory so we're now recreating Route objects if the `spec.host` changed